concourse/scripts/spec/script_testing_helpers.rb

```diff
@@ -0,0 +1,100 @@
+require "aruba/rspec"
+require "pathname"
+
+def find_git_repository_path()
+  path = __FILE__
+  begin
+    if File.exists? File.join(path, '.git')
+      return path
+    end
+    path = File.realpath(File.join(path, '..'))
+  end while path != "/"
+  raise "Cannot find a git repository parent of '#{File.dirname(__FILE__)}'"
+end
+
+def repostory_name()
+  File.basename(find_git_repository_path())
+end
+
+def prepare_aruba_config()
+  RSpec.configure do |config|
+    config.before(:all) do
+      @aruba_orig_workdir = Aruba.config.working_directory
+
+      # We run scripts with git commands, and as we don't want to mess around
+      # with our own repository, we setup the workdir in a different location.
+      #
+      # But Aruba demands it to be relative
+      @workdir = Dir.mktmpdir("paas-cf-tmpdir-")
+      relative_workdir = Pathname.new(@workdir).relative_path_from(Pathname.pwd)
+
+      Aruba.configure do |config|
+        config.working_directory = relative_workdir
+        config.home_directory = @workdir
+        @repo_root = find_git_repository_path()
+
+        config.command_search_paths = [
+          File.join(@repo_root, "concourse", "scripts"),
+          File.join(@repo_root, "manifests", "scripts")
+        ]
+        if ENV["DEBUG"]
+          config.command_launcher = :debug
+        end
+      end
+    end
+    config.after(:all) do
+      # Ensure we clean up after
+      def rmdir(dir)
+        FileUtils.rmdir(dir) if File.exists? dir
+      end
+      rmdir @workdir
+      rmdir @aruba_orig_workdir
+      rmdir @aruba_orig_workdir
+    end
+    config.after(:each) do |example|
+      # Small hack to get the output of the last command in the rspec
+      # failure report. If anybody knows a better solution, please propose it.
+      if example.exception != nil and last_command_started and last_command_started.exit_status != 0
+        raise Exception.new("Last command output: " + last_command_started.output)
+      end
+    end
+  end
+end
+
+def init_concourse_working_directory()
+  @repo_root = find_git_repository_path
+  @repo_name = File.basename(@repo_root)
+
+  # Link the repository directory, as it is done in concourse
+  FileUtils.ln_s(@repo_root, File.join(@workdir,  @repo_name))
+end
+
+
+# Helper to use with aruba to run all commands in a given list or strings
+def run_all(commands)
+  commands.each { |cmd|
+    run_simple(cmd)
+  }
+end
+
+# Helper to run a bash script in verbose mode and capturing
+# all output together
+#
+# Good for scripts and initialise stuff
+#
+def bash_c(script)
+  run_simple("bash -x -c 'exec 2>&1 # Send Stderr to Stdout\n\n#{script}'")
```

Comment:
Should these helpers also run with `set -euo pipefail` so that we know if the setup has failed?